apps/picsa-tools/option-tool/src/app/components/editor/editor.component.ts

```diff
@@ -1,8 +1,91 @@
-import { Component } from '@angular/core';
+import {Component, Inject, OnInit} from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog'
 
 @Component({
   selector: 'option-editor',
   templateUrl: './editor.component.html',
   styleUrls: ['./editor.component.scss'],
 })
-export class EditorComponent {}
+export class EditorComponent implements OnInit {
+  stepCounter: number;
+  practiceEntry: string;
+  gender: string [];
+  benefits: {benefit:string, beneficiary:string[]} [];
+  perfomanceValues: {lowRf:string, midRf:string,  highRf:string};  
+  performanceOptions: string[]=['good','ok','bad']
+  investmentValues: {money:string, time:string};  
+  investmentOptions: string[]=['high','mid','low']
+  benefitsStartTime: string;
+  risk:string;
+
+  constructor(
+    public dialogRef: MatDialogRef<EditorComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any
+  ) { }
+  ngOnInit(): void {
+    this.stepCounter = 1;
+    this.gender=[],
+    this.benefits=[{
+      benefit:'',
+      beneficiary:[]
+    },
+  ]
+  this.perfomanceValues = { lowRf:"ok",midRf:"ok",highRf:"ok"}
+  this.investmentValues = {money:'high', time:'high'}
+  }
+  handlePracticeEntryNext(){
+    this.stepCounter +=1; 
+  }
+  handlePrevious(){
+    if((this.stepCounter-1) > 0){
+      this.stepCounter -=1;
+    }
+  }
+  handleGender(gender:string){
```

Comment:
+ thanks for separating all of these into smaller functions. It will make it much easier if we decide to refactor into separate component files in the future

Reply:
would like to see how we approach this